packages/suite-base/src/players/messageMemoryEstimation.ts

```diff
@@ -206,69 +189,82 @@ export function estimateObjectSize(obj: unknown): number {
+    COMPRESSED_POINTER_SIZE +
+    OBJECT_BASE_SIZE +
+    Array.from(map.entries()).reduce(
+      (acc: number, [key, val]: [unknown, unknown]) =>
+        acc + estimateObjectSize(key) + estimateObjectSize(val),
+      0,
+    );
+
+  const estimateSetSize = (set: Set<unknown>): number =>
+    COMPRESSED_POINTER_SIZE +
+    OBJECT_BASE_SIZE +
+    Array.from(set.values()).reduce(
+      (acc: number, val: unknown) => acc + estimateObjectSize(val),
+      0,
+    );
+
+  const estimateObjectPropertiesSize = (object: Record<string, unknown>): number => {
+    const valuesSize = Object.values(object).reduce(
+      (acc: number, val: unknown) => acc + estimateObjectSize(val),
+      0,
+    );
+    const numProps = Object.keys(obj).length;
+
+    if (numProps > MAX_NUM_FAST_PROPERTIES) {
+      // Dictionary mode estimation for objects with many properties
+      const propertiesDictSize =
+        16 + 5 * 8 + 2 ** Math.ceil(Math.log2((numProps + 2) * 1.5)) * 3 * 4;
+      return (
+        OBJECT_BASE_SIZE + valuesSize + propertiesDictSize - numProps * COMPRESSED_POINTER_SIZE
+      );
+    }
+
+    return OBJECT_BASE_SIZE + valuesSize;
+  };
+
   switch (typeof obj) {
     case "undefined":
-    case "boolean": {
+    case "boolean":
       return SMALL_INTEGER_SIZE;
-    }
-    case "number": {
+
+    case "number":
       return Number.isInteger(obj) ? SMALL_INTEGER_SIZE : HEAP_NUMBER_SIZE;
-    }
-    case "bigint": {
+
+    case "bigint":
       return HEAP_NUMBER_SIZE;
-    }
-    case "string": {
-      // The string length is rounded up to the next multiple of 4.
+
+    case "string":
       return COMPRESSED_POINTER_SIZE + OBJECT_BASE_SIZE + Math.ceil(obj.length / 4) * 4;
-    }
-    case "object": {
+
+    case "object":
       if (Array.isArray(obj)) {
-        return (
-          COMPRESSED_POINTER_SIZE +
-          ARRAY_BASE_SIZE +
-          Object.values(obj).reduce((acc, val) => acc + estimateObjectSize(val), 0)
-        );
-      } else if (ArrayBuffer.isView(obj)) {
+        return estimateArraySize(obj);
+      }
+      if (ArrayBuffer.isView(obj)) {
         return TYPED_ARRAY_BASE_SIZE + obj.byteLength;
-      } else if (obj instanceof Set) {
-        return (
-          COMPRESSED_POINTER_SIZE +
-          OBJECT_BASE_SIZE +
-          Array.from(obj.values()).reduce((acc, val) => acc + estimateObjectSize(val), 0)
-        );
-      } else if (obj instanceof Map) {
-        return (
-          COMPRESSED_POINTER_SIZE +
-          OBJECT_BASE_SIZE +
-          Array.from(obj.entries()).reduce(
-            (acc, [key, val]) => acc + estimateObjectSize(key) + estimateObjectSize(val),
-            0,
-          )
-        );
       }
-
-      let propertiesSize = 0;
-      const numProps = Object.keys(obj).length;
-      if (numProps > MAX_NUM_FAST_PROPERTIES) {
-        // If there are too many properties, V8 stores Objects in dictionary mode (slow properties)
```

Comment:
This comment is not usefull anymore? I thoug it was nice to keep at least the medium article if we need to see it in the future

Reply:
Yeah it could be useful in the future I'll add back the explanation